Bump glimmer-vm packages to 0.24.

chancancode opened the pull request:
* [x] ESLint (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=ESLint)
* [x] Static Content (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=static%20content)
* [x] Dynamic Content (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=dynamic%20content)
* [x] Syntax (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=syntax)
* [x] Helpers (http://localhost:5200/tests/index.html?hidepassed&package=ember-glimmer&filter=helpers)
    * [x] Custom Helpers (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=custom%20helpers)
    * [x] {{if}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bif%7D%7D)
    * [x] {{unless}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bunless%7D%7D)
    * [x] {{input}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%3A%20%7B%7Binput)
    * [x] {{textarea}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Btextarea%7D%7D)
    * [x] {{partial}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bpartial%7D%7D)
    * [x] {{yield}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Byield%7D%7D)
    * [x] {{hash}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bhash%7D%7D)
    * [x] {{concat}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bconcat%7D%7D)
    * [x] {{log}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Blog%7D%7D)
    * [x] {{loc}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bloc%7D%7D)
    * [x] {{get}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bget%7D%7D)
    * [x] {{render}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Brender%7D%7D)
    * [x] {{mut}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bmut%7D%7D)
    * [x] {{readonly}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Breadonly%7D%7D)
    * [x] closure {{action}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=closure%20%7B%7Baction%7D%7D)
    * [x] element action (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=element%20action)
* [x] Components (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=Components%20test&hidepassed)
    * [x] this.attrs, positional arguments
        * We broke this in https://github.com/emberjs/ember.js/commit/d979a63f3fb9521a5ae6ae46fef017b8868ab580
    * [x] Cascading failures (cannot reuse the same DIV, cannot start nested transactions, etc)
    * [x] contextual components (http://localhost:7200/tests/index.html?hidepassed&filter=contextual)
    * [x] local lookup (http://localhost:7200/tests/index.html?hidepassed&filter=local)
* [x] Iterable (http://localhost:5200/tests/index.html?package=ember-glimmer&module=Iterable)
    * Remove? (https://github.com/emberjs/ember.js/pull/14763#issuecomment-301955781)
* [x] ...

chancancode committed:
Bump Glimmer version

chancancode committed:
Fix build scripts for latest Glimmer

chancancode committed:
These are built-in now

chancancode committed:
Fix macro syntax

chancancode committed:
Update helpers to use new args

chancancode committed:
✂️ symbolTable part 1

chancancode committed:
Fix iterable hook

chancancode committed:
Fix compilation

chancancode committed:
✂️ getArgs

chancancode committed:
Fix macros

chancancode committed:
Fix closure component helper

chancancode committed:
[WIP] Fix component args

This will break a lot of things but is a good excuse to rethink the
"process args" code.

chancancode committed:
Fix {{hash}}

homu commented:
:umbrella: The latest upstream changes (presumably #15247) made this pull request unmergeable. Please resolve the merge conflicts.

Commit:
Merge branch 'master' into bump-glimmer

Commit:
Fix `{{mut}}`.

Commit:
Fix `(action`.

Commit:
Fix `{{action`.

Commit:
Fix non-singleton `{{render` tests.

homu commented:
:umbrella: The latest upstream changes (presumably #15254) made this pull request unmergeable. Please resolve the merge conflicts.

Commit:
Ensure managers return `null` from `prepareArgs` to "pass through".

Commit:
Make test helpers clear up renderers upon teardown.

This is only needed when a catastrophic failure happens that
prevents the normal removal from working. Doing this in the
teardown greatly reduces the number of invalid faulures when
errors occur during rendering.

Commit:
fixup! Fix non-singleton `{{render` tests.

Commit:
Initial pass at supporting "rest arg" style positionals.

Commit:
Remove iterableForObject tests.

These are largely duplication functionality in `{{each-in}}`'s own
tests.  They were added as a way to flesh out the iterable implementation.

Commit:
Update glimmer versions to 0.24.0-alpha.2.

Commit:
Merge remote-tracking branch 'origin/master' into bump-glimmer

Commit:
Split component managers out from syntax definition.

knownasilya commented:
Interesting seeing the dog fooding of component-managers and how they are used. Good stuff and much to learn.

Commit:
Fix closure component currying and positional param munging.

Commit:
Fixup layout caching tests.

rwjblue commented on packages/ember-glimmer/lib/component-managers/curly.js:
```diff
@@ -103,13 +96,65 @@ class CurlyComponentLayoutCompiler {
 
 CurlyComponentLayoutCompiler.id = 'curly';
 
+export class PositionalArgumentReference {
```
Need to conform to the `PathReference` interface (missing `get` at least ATM).

Commit:
Fix issues with AST transform for `{{this.attrs.foo}}` -> `{{@foo}}`.

Commit:
Bump glimmer-vm packages to 0.24.0-beta.1.

Commit:
Ensure that newly created `PositionalArgumentsReference` implements `PathReference` API.

Commit:
Fix `ember-engines-mount-params` feature with glimmer-vm@0.24.

Commit:
Update to glimmer-vm 0.24.0-beta.2.

Commit:
Fix `glimmer-custom-component-manager` feature for glimmer-vm 0.24.x.

Commit:
Bump glimmer-vm packages to 0.24.0-beta.3.

Fixes a few node tests.

rwjblue commented:
Did some quick asset size checks, and the `min + gz` size is slightly better (current `master` `120.54 KB`; after this PR `120.15 KB`).

Commit:
Merge remote-tracking branch 'origin/master' into bump-glimmer

Commit:
Bump glimmer-vm packages to 0.24.0-beta.4

rwjblue commented:
I did some general benchmarking (against the emberaddons.com app), and it seems that we haven't introduced a regression here.  

![phases](https://cloud.githubusercontent.com/assets/12637/26511603/d0a5fd3a-4230-11e7-81ff-f8653683a965.png)

Seems good to ship...

knownasilya commented:
:shipit:

rwjblue commented:
Did another test against emberobserver.com (which has a larger rendering phase):

![phases](https://cloud.githubusercontent.com/assets/12637/26523054/e063283c-42db-11e7-8914-dbd4b671aaa5.png)

chadhietala commented on packages/ember-glimmer/lib/component-managers/outlet.js:
```diff
@@ -0,0 +1,144 @@
+    }
+  }
+
+  didCreateElement() {}
```
Can't these go on the abstract manager?

chadhietala commented on packages/ember-glimmer/lib/component-managers/render.js:
```diff
@@ -0,0 +1,111 @@
+    return null;
+  }
+
+  didCreateElement() {}
```
Same comment about putting them on the abstract manager.

rwjblue replied:
Yep, totally agreed.

Commit:
Migrate default component manager hooks to abstract base class.

chadhietala commented:
LGTM

knownasilya commented:
Woop
